### PR TITLE
fix issue 17102 - Don't pass null to strlen in std.file.cenforce

### PR DESCRIPTION
**Pull request by Burgos**
https://issues.dlang.org/show_bug.cgi?id=17102

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -179,7 +179,7 @@ private T cenforce(T)(T condition, const(char)[] name, const(FSChar)* namez,
         import core.stdc.wchar_ : wcslen;
         import std.conv : to;
 
-        auto len = wcslen(namez);
+        auto len = namez? wcslen(namez) : 0;
```
Comment (JackStouffer) [inline std/file.d]: space around `?`

File: std/file.d [hunk 2]
```diff
@@ -196,12 +196,23 @@ private T cenforce(T)(T condition, const(char)[] name, const(FSChar)* namez,
     {
         import core.stdc.string : strlen;
 
-        auto len = strlen(namez);
+        auto len = namez? strlen(namez) : 0;
```
Comment (JackStouffer) [inline std/file.d]: ditto

**Comment by Burgos** [comment]
> Only the windows version?

Huh, I thought I fixed both, let me see.

**Comment by Burgos** [comment]
Posix version:
https://github.com/dlang/phobos/pull/5049/files#diff-52580fb75b304ba7b04a6b178fe6cdf4R199

Windows version:
https://github.com/dlang/phobos/pull/5049/files#diff-52580fb75b304ba7b04a6b178fe6cdf4R170

**Comment by Burgos** [comment]
Updated with spaces around `?`.

**Comment by Burgos** [comment]
Updated.

**Pull request merged**